lib/conflisp/evaluator.rb

```diff
@@ -35,14 +32,20 @@ def resolve(expression) # rubocop:disable Metrics/MethodLength
       else
         expression
       end
-    end
-
-    def fn_defined?(fn_name)
-      registry.key?(fn_name)
+    rescue Conflisp::ConflispError => e
+      e.conflisp_stack << expression
+      raise e
     end
 
     def apply(fn_name, *args)
-      instance_exec(*args, &registry[fn_name])
+      method = registry[fn_name]
+      raise MethodMissing, fn_name unless method
```

Comment:
Should this be `Conflisp:: MethodMissing`?

Reply:
technically no, but it's really confusing. I had to put `Conflisp::RuntimeError` because rubocop was complaining, but it is really confusing, so I will prefix everything